src/index.ts

```diff
@@ -294,9 +294,11 @@ export async function main(
       .filter(Boolean)
       .map(evar => `${evar}=${process.env[evar] || ''}\n`)
       .join('')
-    uploadFile = uploadFile.concat(vars).concat(MARKER_ENV_END)
+    uploadFileChunks.push(Buffer.from(vars))
+    uploadFileChunks.push(Buffer.from(MARKER_ENV_END))
   }
 
+  const uploadFile = Buffer.concat(uploadFileChunks)
```

Comment:
This performs a final allocation to concatenate the array of Buffer instances.[^1]

[^1]: [Buffer | Node.js v16.15.0 Documentation]

[Buffer | Node.js v16.15.0 Documentation]: https://nodejs.org/docs/latest-v16.x/api/buffer.html#static-method-bufferconcatlist-totallength